databroker/core.py

```diff
@@ -499,6 +487,15 @@ def canonical(self, *, fill):
             copies of the documents.
 
         """
+        # Special case for 'delayed' since it *is* supported in the local mode
+        # of usage.
+        if fill == 'delayed':
+            raise NotImplementedError(
+                "Delayed access is not yet supported via the client--server "
+                "usage.")
+        FILL_OPTIONS = {'yes', 'no'}
+        if fill not in FILL_OPTIONS:
+            raise ValueError(f"Options for 'fill' are {FILL_OPTIONS}")
```

Comment:
I think it's worth to also print the value of `fill` as well to understand what we are hitting.

Reply:
fixed